### ISSUE TITLE
Curious about adding support for `torch.onnx.export`

### ISSUE THREAD

**Issue by slitcch** [LaurentMazare/tch-rs#588]
Hi! I'm a machine learning practitioner who's very sick of Python looking for greener pastures. For us, we need to be able to export trained models to ONNX files so we can run them in our inference pipeline. What would it take to add "first-class" support to tch-rs for ONNX export?

**Comment by John0x** [comment]
Direct onnx support would be nice, but it looks like it's already possible with a workaround.
See: https://github.com/LaurentMazare/tch-rs/issues/311

**Comment by John0x** [comment]
> > Direct onnx support would be nice, but it looks like it's already possible with a workaround. See: #311
> 
> Hi it mentions using `tch::CModule::create_by_tracing` to do this, but when to use this method and how can we export to torchscript by using this ?

See: https://github.com/LaurentMazare/tch-rs/tree/main/examples/jit-trace

You can trace it in rust, load it in pytorch and then export that to onnx. Haven't tried it, but it looks like that should work.

**Comment by LaurentMazare** [comment]
I would agree with @John0x : you can trace the module in Rust and export it using the code from the example, and then you will have to load it from Python via `torch.load` and export it to onnx if that's the final format you want.
There is currently no onnx export in libtorch, see [pytorch#17869](https://github.com/pytorch/pytorch/issues/17869) for details. If this gets added on the c++ side, it will hopefully be easy to add to this crate.

**Comment by mdrokz** [comment]
> > > Direct onnx support would be nice, but it looks like it's already possible with a workaround. See: #311
> > 
> > 
> > Hi it mentions using `tch::CModule::create_by_tracing` to do this, but when to use this method and how can we export to torchscript by using this ?
> 
> See: https://github.com/LaurentMazare/tch-rs/tree/main/examples/jit-trace
> 
> You can trace it in rust, load it in pytorch and then export that to onnx. Haven't tried it, but it looks like that should work.

Ah this makes sense thanks

**Comment by LaurentMazare** [comment]
Closing this for now, we may revisit if/when onnx export makes it into libtorch.
